core/templates/dev/head/pages/exploration-player-page/learner-experience/learner-answer-info-card.directive.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for asking learner for answer details.
+ */
+
+require('domain/utilities/UrlInterpolationService.ts');
+require('pages/exploration-player-page/services/player-transcript.service.ts');
+require(
+  'pages/exploration-player-page/services/learner-answer-info.service.ts');
+require('pages/exploration-player-page/services/exploration-engine.service.ts');
+require('services/ExplorationHtmlFormatterService.ts');
+
+angular.module('oppia').directive('learnerAnswerInfoCard', [
+  'UrlInterpolationService', function(UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {},
+      bindToController: {
+        submitAnswer: '&'
+      },
+      templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+        '/pages/exploration-player-page/learner-experience/' +
+        'learner-answer-info-card.directive.html'),
+      controllerAs: '$ctrl',
+      controller: [
+        'ExplorationEngineService', 'ExplorationHtmlFormatterService',
+        'LearnerAnswerInfoService', 'PlayerTranscriptService',
+        function(ExplorationEngineService, ExplorationHtmlFormatterService,
+            LearnerAnswerInfoService, PlayerTranscriptService) {
+          var ctrl = this;
+          ctrl.answerDetails = null;
+          var interaction = ExplorationEngineService.getState().interaction;
+          ctrl.submitLearnerAnswerInfo = function() {
+            LearnerAnswerInfoService.recordLearnerAnswerInfo(
+              ctrl.answerDetails);
+            PlayerTranscriptService.addNewInput(
+              {answerDetails: ctrl.answerDetails}, false);
+            PlayerTranscriptService.addNewResponse(
+              '<p translate="I18N_SOLICIT_ANSWER_DETAILS_FEEDBACK"></p>');
```

Comment:
Again, no JS in HTML please.

One other note, too: the language in which this is shown to the user should probably match the language of the exploration, since it's appearing in that context (you can use English if we don't have the translation of this string available in that language, though). Note that the "translate" attribute in Angular only matches the chosen language of the user, which is not the same thing.

Reply:
Removed from JS